packages/relayer-lib/src/tx_builder/eth_to_cosmos.rs

```diff
@@ -399,26 +344,115 @@ where
                 serde_json::to_string(&header)?
             );
             headers.push(header);
-            latest_trusted_slot = finality_update.finalized_header.beacon.slot;
         }
 
-        let number_of_finality_updates = headers.len() - number_of_period_updates;
-        let initial_period = ethereum_client_state
-            .compute_sync_committee_period_at_slot(ethereum_consensus_state.slot);
-        let latest_period =
-            ethereum_client_state.compute_sync_committee_period_at_slot(latest_trusted_slot);
-
-        let proof_block_number = headers
+        let (proof_block_number, proof_slot) = headers
             .last()
             .map(|header| {
-                header
-                    .consensus_update
-                    .finalized_header
-                    .execution
-                    .block_number
+                (
+                    header
+                        .consensus_update
+                        .finalized_header
+                        .execution
+                        .block_number,
+                    header.consensus_update.finalized_header.beacon.slot,
+                )
             })
             .ok_or_else(|| anyhow::anyhow!("No headers found"))?;
 
+        Ok((proof_block_number, proof_slot, headers))
```

Comment:
I feel like we should only be dealing with `proof_slot` and not `proof_block_number ` since `proof_block_number` is only used in `inject_ethereum_proofs`. So couldn't we do the `slot -> height` conversion there?